Algorithm/QCAlgorithm.cs

```diff
@@ -1284,8 +1322,54 @@ public Option AddOption(string underlying, Resolution resolution = Resolution.Mi
             Universe universe;
             if (!UniverseManager.TryGetValue(canonicalSymbol, out universe))
             {
-                var settings = new UniverseSettings(resolution, leverage, false, false, TimeSpan.Zero);
-                universe = new OptionChainUniverse(canonicalSecurity, settings, SecurityInitializer);
+                var settings = new UniverseSettings(resolution, leverage, true, false, TimeSpan.Zero);
+                universe = new OptionChainUniverse(canonicalSecurity, settings, SubscriptionManager, SecurityInitializer);
+                UniverseManager.Add(canonicalSymbol, universe);
+            }
+
+            return canonicalSecurity;
+        }
+
+        /// <summary>
+        /// Creates and adds a new <see cref="Future"/> security to the algorithm
+        /// </summary>
+        /// <param name="symbol">The futures contract symbol</param>
+        /// <param name="resolution">The <see cref="Resolution"/> of market data, Tick, Second, Minute, Hour, or Daily. Default is <see cref="Resolution.Minute"/></param>
+        /// <param name="market">The futures market, <seealso cref="Market"/>. Default is value null and looked up using BrokerageModel.DefaultMarkets in <see cref="AddSecurity{T}"/></param>
+        /// <param name="fillDataForward">If true, returns the last available data even if none in that timeslice. Default is <value>true</value></param>
+        /// <param name="leverage">The requested leverage for this equity. Default is set by <see cref="SecurityInitializer"/></param>
+        /// <returns>The new <see cref="Future"/> security</returns>
+        public Future AddFuture(string symbol, Resolution resolution = Resolution.Minute, string market = null, bool fillDataForward = true, decimal leverage = 0m)
+        {
+            if (market == null)
+            {
+                if (!BrokerageModel.DefaultMarkets.TryGetValue(SecurityType.Future, out market))
+                {
+                    throw new Exception("No default market set for security type: " + SecurityType.Future);
+                }
+            }
+
+            Symbol canonicalSymbol;
+            var alias = "?" + symbol;
```

Comment:
Good point. Will check it. I left ?SPY convention for the sake of convenience. Probably we could use different symbol for futures... e.g. "/" like many do.